Fail to build when ENABLE_COMPUTED_GOTO is turned off

By default, RV32C support is enabled. When the configuration `ENABLE_COMPUTED_GOTO` is set to `0`, it fails to build:
```c
$ make ENABLE_COMPUTED_GOTO=0
  CC	map.o
  CC	riscv.o
...
riscv.c:1417:19: error: lvalue required as left operand of assignment
 1417 |             index = (inst & INST_6_2) >> 2;
      |                   ^
riscv.c:1420:39: error: array subscript is not an integer
 1420 |             TABLE_TYPE op = jump_table[index];
      |                                       ^
...
```

--- a/riscv.c
+++ b/riscv.c
@@ -1414,28 +1414,32 @@ void rv_step(struct riscv_t *rv, int32_t cycles)
 
         // standard uncompressed instruction
         if ((inst & 3) == 3) {
-            index = (inst & INST_6_2) >> 2;
+            uint32_t index = (inst & INST_6_2) >> 2;
 
             // dispatch this opcode
             TABLE_TYPE op = jump_table[index];
             assert(op);
+            if (!op(rv, inst))
+                break;
             rv->inst_len = INST_32;
 
         } else {
             // standard compressed instruction
+            inst &= 0x0000FFFF;
             const uint16_t c_index =
-                (inst & FR_C_15_13 >> 11) | (inst & FR_C_1_0);
+                (inst & FC_FUNC3 >> 11) | (inst & FC_OPCODE);
 
             // dispactch c_opcode (compressed instructions)
-            TABLE_TYPE_RVC op = jump_tablec[c_index];
+            TABLE_TYPE_RVC op = jump_table_rvc[c_index];
             assert(op);
+            if (!op(rv, inst))
+                break;
             rv->inst_len = INST_16;
         }
-
-        if (!op(rv, inst))
-            break;
+        
         // increment the cycles csr
         rv->csr_cycle++;
+    }
 #endif  // ENABLE_COMPUTED_GOTO
 }
 

